ngraph/core/src/op/util/convert_color_nv12_base.cpp

```diff
@@ -0,0 +1,231 @@
+    } else {
+        const auto& uv_tensor = input_values[1];
+        ngraph::runtime::reference::color_convert_nv12(y_tensor->get_data_ptr<ET>(),
+                                                       uv_tensor->get_data_ptr<ET>(),
+                                                       output_value->get_data_ptr<ET>(),
+                                                       batch_size,
+                                                       image_h,
+                                                       image_w,
+                                                       image_w * image_h,
+                                                       image_w * image_h / 2,
+                                                       color_format);
+    }
+    return true;
+}
+
+bool evaluate_nv12_convert(const ov::HostTensorVector& input_values,
+                           const ov::HostTensorPtr& output_value,
+                           bool single_tensor,
+                           ov::op::util::ConvertColorNV12Base::ColorConversion conv_format) {
+    bool rc = false;
+    switch (input_values[0]->get_element_type()) {
+        NGRAPH_TYPE_CASE(evaluate_nv12_convert, u8, input_values, output_value, single_tensor, conv_format);
+        NGRAPH_TYPE_CASE(evaluate_nv12_convert, f16, input_values, output_value, single_tensor, conv_format);
+        NGRAPH_TYPE_CASE(evaluate_nv12_convert, bf16, input_values, output_value, single_tensor, conv_format);
+        NGRAPH_TYPE_CASE(evaluate_nv12_convert, f32, input_values, output_value, single_tensor, conv_format);
+        NGRAPH_TYPE_CASE(evaluate_nv12_convert, f64, input_values, output_value, single_tensor, conv_format);
+    default:
+        break;
+    }
+    return rc;
+}
+
+}  // namespace color_convert_nv12_op
+
+bool ov::op::util::ConvertColorNV12Base::visit_attributes(AttributeVisitor& visitor) {
+    return true;
+}
+
+bool ov::op::util::ConvertColorNV12Base::evaluate(const HostTensorVector& output_values,
+                                                  const HostTensorVector& input_values) const {
+    NGRAPH_OP_SCOPE(v0_ConvertColorNV12_evaluate);
+    OPENVINO_ASSERT(ngraph::validate_host_tensor_vector(output_values, 1));
+    NODE_VALIDATION_CHECK(this,
+                          get_input_size() == 1 || get_input_size() == 2,
+                          "NV12 conversion shall have one or 2 inputs, but it is ",
+                          get_input_size());
+    auto single_plane = get_input_size() == 1;
+    return color_convert_nv12_op::evaluate_nv12_convert(input_values, output_values[0], single_plane, m_format);
+}
+
+bool ov::op::util::ConvertColorNV12Base::has_evaluate() const {
+    NGRAPH_OP_SCOPE(v0_ConvertColorNV12Base_has_evaluate);
+
+    return is_type_supported(get_input_element_type(0));
+}
+
+bool ov::op::util::ConvertColorNV12Base::is_type_supported(const ov::element::Type& type) const {
+    return type.is_dynamic() || type.is_real() || type == ov::element::u8;
```

Comment:
Why don't we support integer types?

Reply:
I checked this case with @jane-intel while reviewing 'operation specification'. My original proposal was to support only u8 & fp32, and it was updated to 'u8 or any supported floating point type'. See PR #7595

Reply:
@jane-intel can you please comment on this? User can do 'convert_element_type' if needed, but I'm also ok to update just approved specification and update to 'u8 + any_float + any_number_16bit_or_bigger'

Reply:
I was convinced that we could decode NV12 only to the list of types we listed in the spec. but if there is another use case -- then let's all them to spec and here also

Reply:
Currently specification states "u8 or any supported floating-point type." - here is checked exactly this (+ dynamic type for evaluation on inference stage)